octodns/record/__init__.py

```diff
@@ -757,6 +758,9 @@ def validate(cls, data, _type):
             reasons.append('empty value')
         elif not data:
             reasons.append('missing value')
+        elif not FQDN(data, allow_underscores=True).is_valid:
+            reasons.append('{} value "{}" is not a valid FQDN'
+                           .format(_type, data))
```

Comment:
Does the `FQDN` method provide any information about what's wrong with the value? It might be nice (if it's easy) to provide what the user needs to do it fix the problem. If it's not easy or doesn't support it I think this is still ship-able.

Reply:
Looks like it doesn't have that rich information, just true or false about if the string is valid:

https://fqdn.readthedocs.io/en/latest/